### PR TITLE
Add username env

### PR DESCRIPTION
**Pull request by fbrissi**
#1 
@dbpolito suggested add only env to username

### CODE REVIEW

**Comment by fabriciojs** [comment]
@fbrissi so you want this to be merged, and close out #1 - is that the case?

The nice thing I liked about the first option was the ability to setup several users at once, which we don't have here, right...

I am fine either way, I guess leaning a bit to the first implementation.

**Comment by fbrissi** [comment]
@fabriciojs I like #1, but @dbpolito suggested an alternative to simplify the logic adding user by env, so I opened this suggestion to compare, but I think it's a good idea we can add many users.

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,9 @@
 FROM alpine:3.12
 
 ENV AUTHORIZED_KEYS=""
+ENV SSH_USERNAME="kool"
 
-RUN adduser -D -u 1337 kool && \
-    passwd -u kool && \
```
Comment (dbpolito) [inline Dockerfile:6]: i think i had to add this for a reason, did you test without it?

File: README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ By default it has a user `kool` and can only be logged using SSH Keys by:
 ```bash
 AUTHORIZED_KEYS=$(cat authorized_keys.txt)
 
-docker run --rm --init -p22:22 -e AUTHORIZED_KEYS=${AUTHORIZED_KEYS} kooldev/sshd
+docker run --rm --init -p22:22 -e AUTHORIZED_KEYS=${AUTHORIZED_KEYS} -e SSH_USERNAME='kool' kooldev/sshd
 ```
```
Comment (dbpolito) [inline README.md:20]: i think we can improve the readme a bit... maybe a separate section for: `Custom SSH User`

File: entrypoint [hunk 3]
```diff
@@ -1,16 +1,22 @@
 #!/bin/sh
-
 set -e
 
-if [ ! -z "$AUTHORIZED_KEYS" ]; then
-    echo "$AUTHORIZED_KEYS" > /home/kool/.ssh/authorized_keys
-elif [ -f "/authorized_keys" ]; then
-    cp /authorized_keys /home/kool/.ssh/authorized_keys
+if [ ! -d /home/"${SSH_USERNAME}"/.ssh ]; then
```
Comment (dbpolito) [inline entrypoint:4]: what if you set `SSH_USERNAME=root`, the user exists but the folder doesn't?  Maybe we should check if user exists and create, and the rest i think it's safe to always execute

Comment (dbpolito) [inline entrypoint:4]: also i think you can remove the quotes `"${SSH_USERNAME}"`, the quotes are there on the AUTHORIZED_KEYS for break lines, not the case for user

**Review by dbpolito — CHANGES_REQUESTED** [review]

**Comment by dbpolito** [comment]
I'm not against https://github.com/kool-dev/docker-sshd/pull/1 but i think most of the cases you wont need it... so i prefer treating edge cases differently than introducing too much complexity here.

So i prefer this one.

**Comment by dbpolito** [comment]
Also it breaks the usage of this, we support mounting file or env vars, on the https://github.com/kool-dev/docker-sshd/pull/1 approach we only support custom users by ENV.

**Comment by dbpolito** [comment]
Thanks for this, but let's move with https://github.com/kool-dev/docker-sshd/pull/1 approach.

**Pull request closed without merging**